homeassistant/components/southern_company/sensor.py

```diff
@@ -0,0 +1,305 @@
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .config_flow import InvalidAuth
+from .const import DOMAIN as SOUTHERN_COMPANY_DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+ICON = "mdi:currency-usd"
+
+
+SENSORS: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key="dollars_to_date",
+        name="Monthly cost",
+        device_class=SensorDeviceClass.MONETARY,
+        state_class=SensorStateClass.TOTAL,
+    ),
+    SensorEntityDescription(
+        key="total_kwh_used",
+        name="Monthly consumption",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key="average_daily_cost",
+        name="Average daily cost",
+        device_class=SensorDeviceClass.MONETARY,
+        state_class=SensorStateClass.TOTAL,
+    ),
+    SensorEntityDescription(
+        key="average_daily_usage",
+        name="Average daily usage",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key="projected_usage_high",
+        name="Higher projected monthly usage",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key="projected_usage_low",
+        name="Lower projected monthly usage",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    SensorEntityDescription(
+        key="projected_bill_amount_low",
+        name="Lower projected monthly cost",
+        device_class=SensorDeviceClass.MONETARY,
+        state_class=SensorStateClass.TOTAL,
+    ),
+    SensorEntityDescription(
+        key="projected_bill_amount_high",
+        name="Higher projected monthly cost",
+        device_class=SensorDeviceClass.MONETARY,
+        state_class=SensorStateClass.TOTAL,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Southern Company sensor."""
+
+    southern_company_connection = hass.data[SOUTHERN_COMPANY_DOMAIN][entry.entry_id]
+
+    coordinator = SouthernCompanyCoordinator(hass, southern_company_connection)
+    entities: list[SouthernCompanySensor] = []
+    await southern_company_connection.get_jwt()
+    for account in await southern_company_connection.get_accounts():
+        device = DeviceInfo(
+            identifiers={((SOUTHERN_COMPANY_DOMAIN), account.number)},
```

Comment:
domain shouldn't be wrapped in a tuple.

```suggestion
            identifiers={(DOMAIN, account.number)},
```